core/src/Kokkos_Core.hpp

```diff
@@ -274,6 +274,23 @@ class ScopeGuard {
 
 }  // namespace Kokkos
 
+namespace Kokkos {
+// Partitioning an Execution Space: expects space and integer arguments for
+// relative weight
+//   Customization point for backends
+//   Default behavior is to return the passed in instance
+template <class ExecSpace, class... Args>
+std::array<ExecSpace, sizeof...(Args)> partition_space(ExecSpace space,
```

Comment:
Are we intentionally requiring people to partition into a compile-time sized set of args? Should we have a variant where this takes a vector of weights and returns a vector of instances?

Reply:
Hm interesting question, we could have both, that said I didn't see a real need for that much runtime choice.

Reply:
The partition count in EMPIRE will be runtime-determined

Reply:
That said, it could just call the current unweighted CUDA implementation that just makes separate streams one by one, but that's probably not the usage you were going for.

Reply:
done, using std::vector now